### DISCUSSION TITLE
Is it possible to get the angle of rotation of the text area via PaddleOCR?

### DISCUSSION THREAD

**Discussion by Aflexg** [PaddlePaddle/PaddleOCR#13264] [Q&A] [answered]
PaddleOCR can correctly recognize 90, 180 and even 270 degree rotated text in a mode use_angle_cls=True, but it doesnt provide any information about the angle of rotation in a  result.As a result, area with a text, cropped by provided coordinates usually has incorrect orientation.Can i get the information about the angle of rotation of the text area somehow to get correctly orientated area witht text ?

**Comment by UserWangZz** [comment]
Specifically you need to look at the original code to determine the output of the classification model.

**Reply by Aflexg** [reply]
To be honest i cant understand how does this code work for now.I am compeltly sure that it must rotate part of the image with a text  if it is not horizontal but i cant find any image rotation operations in the code.

![image](https://github.com/PaddlePaddle/PaddleOCR/assets/153363231/d534d0d0-1a3b-4bc3-9897-a652cfedb33f)

**Reply by UserWangZz** [reply, marked as answer]
You can see this part code in paddleocr.py, if you use paddleocr api. About 755 line, use a classifier model to det the det model's result. and then you can find second part code in tools/infer/predict_cls.py. Here use cls model to rectify det box.
https://github.com/PaddlePaddle/PaddleOCR/blob/7a3c580909be25ca3c560ede560101d4f786d155/paddleocr.py#L725-L762

https://github.com/PaddlePaddle/PaddleOCR/blob/7a3c580909be25ca3c560ede560101d4f786d155/tools/infer/predict_cls.py#L119-L126

**Reply by Aflexg** [reply]
Thank you very much for the provided information. With your help, I was able to get all the necessary information about the rotation angles of the text areas in the image.

**Reply by Aagam2** [reply]
Hello @Aflexg how did you get information about angles besides 0 and 180. In the model configuration when I specify label_list=['0', '90', '180', '270']
The model's results are not correct.

**Reply by Aflexg** [reply]
Hello @Aagam2. I’m not sure that I get information about angles other than 0 and 180. All I did was make minor modifications to the paddleocr.py script so that it outputs data about rotation angles using the information mentioned above.

**Reply by Aagam2** [reply]
Yes I was able to get information about the angles after making modification but it is only working for 0 and 180 degree rotated image.
Thank you for the reply.

**Reply by UserWangZz** [reply]
> Hello @Aagam2. I’m not sure that I get information about angles other than 0 and 180. All I did was make minor modifications to the paddleocr.py script so that it outputs data about rotation angles using the information mentioned above.

Hello, the cls model in paddleocr only cls [0, 180] two angles. if you have more requirment about cls model, just train a cls model

**Reply by Aagam2** [reply]
Understood. Thank you for the reply.
